engine/graphics/src/graphics_adapter.h

```diff
@@ -275,9 +278,122 @@ namespace dmGraphics
         IsExtensionSupportedFn m_IsExtensionSupported;
         GetNumSupportedExtensionsFn m_GetNumSupportedExtensions;
         GetSupportedExtensionFn m_GetSupportedExtension;
-        IsMultiTargetRenderingSupportedFn m_IsMultiTargetRenderingSupported;
+        GetNumTextureHandlesFn m_GetNumTextureHandles;
         GetPipelineStateFn m_GetPipelineState;
+        IsContextFeatureSupportedFn m_IsContextFeatureSupported;
     };
+
+    #define DM_REGISTER_GRAPHICS_FUNCTION(tbl, adapter_name, fn_name) \
+        tbl.m_##fn_name = adapter_name##fn_name
+    #define DM_REGISTER_GRAPHICS_FUNCTION_TABLE(tbl, adapter_name) \
+        DM_REGISTER_GRAPHICS_FUNCTION(tbl, adapter_name, NewContext); \
+        DM_REGISTER_GRAPHICS_FUNCTION(tbl, adapter_name, DeleteContext); \
+        DM_REGISTER_GRAPHICS_FUNCTION(tbl, adapter_name, Initialize); \
+        DM_REGISTER_GRAPHICS_FUNCTION(tbl, adapter_name, Finalize); \
+        DM_REGISTER_GRAPHICS_FUNCTION(tbl, adapter_name, GetWindowRefreshRate); \
+        DM_REGISTER_GRAPHICS_FUNCTION(tbl, adapter_name, OpenWindow); \
+        DM_REGISTER_GRAPHICS_FUNCTION(tbl, adapter_name, CloseWindow); \
```

Comment:
This setup is done so that we can easily detect missing functions from the adapter implementations. Previously we never checked that but now an adapter has to implement all these functions that are required by the adapter "interface"